src/main/java/org/checkerframework/checker/noliteral/NoLiteralVisitor.java

```diff
@@ -0,0 +1,230 @@
+package org.checkerframework.checker.noliteral;
+
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.Tree;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.ExecutableElement;
+import org.checkerframework.checker.noliteral.qual.MaybeConstant;
+import org.checkerframework.checker.noliteral.qual.NonConstant;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.common.basetype.BaseTypeVisitor;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedExecutableType;
+import org.checkerframework.framework.type.AnnotatedTypeParameterBounds;
+import org.checkerframework.framework.type.TypeHierarchy;
+import org.checkerframework.javacutil.ElementUtils;
+
+/**
+ * This visitor has a lot of code in it that permits this type system to operate as expected with
+ * the bottom type as its default. Most of that code is either from or inspired by AWS' Data
```

Comment:
Please fix pluralization.

Reply:
I rewrote this so that the possessive could be removed, but this is actually an interesting grammar corner case. AWS is an initialism, so you would normally using 's when turning it into its possessive form. However, the S in AWS stands for a plural noun ("Services"), so if you wrote it out, it would be "Amazon Web Services'". I'm actually not sure what's correct: a (very good use of time) 10 minute google search turned up multiple results that give each answer to similar questions, though I think there's a bit of a bias toward "AWS's" rather than "AWS'", it's certainly not settled.